### PR TITLE
Make RecipeWrapper implement RecipeInput

### LINKED ISSUE #1143: RecipeWrapper should extend RecipeInput
**Minecraft Version:** 1.21

**NeoForge Version:** 21.017

**Description of issue:**

`RecipeWrapper` should implement `RecipeInput`. As it is right now it can't be used for `RecipeManager.getRecipeFor` anymore.

### PR DESCRIPTION
**Pull request by TelepathicGrunt**
With changes to 1.21 recipes, Mojang has them no longer implementing Container. Instead they implement RecipeInput which only requires modders to define two methods. 

Arguments were made to keep recipewrapper that were convincing imo. Did switch to keeping it but implement recipeInput

closes https://github.com/neoforged/NeoForge/issues/1143

### CODE REVIEW

**Comment by Shadows-of-Fire** [comment]
Many people use ItemStackHandler without needing to subclass it.  If you want to drop this class, we should make ISH directly implement RecipeInput

**Comment by TelepathicGrunt** [comment]
@Shadows-of-Fire Yeah not a problem. Basically like this? Was a clean change to do
https://github.com/neoforged/NeoForge/pull/1149/commits/947e90c7b31e547e8da418a71cfafb484f7ffe62

**Comment by Technici4n** [comment]
Hmmm I don't like mixing generic transfer stuff and recipe specifics.

**Comment by Technici4n** [comment]
Maybe RecipeWrapper should not be deleted, but instead implement `RecipeInput`?

**Comment by TelepathicGrunt** [comment]
> Maybe RecipeWrapper should not be deleted, but instead implement RecipeInput?

@Technici4n RecipeInput is just two methods. You save no work by using RecipeWrapper over RecipeInput if RecipeWrapper is kept. You still have to implement the two methods. RecipeWrapper should stay dead since you should use RecipeInput instead. But I can undo Shadow’s suggestion then

**Comment by Technici4n** [comment]
Wouldn't `RecipeWrapper` just be a simple `size` -> `getSlots` and `getItem` -> `getStackInSlot` bouncer? It's not much added value but it's nonzero. Given how annoying `Recipe` is already to implement, I don't think we need this small amount of "convenience".

TLDR: I'm fine with the PR as is.

File: src/main/java/net/neoforged/neoforge/items/wrapper/RecipeWrapper.java [hunk 1]
```diff
@@ -5,12 +5,11 @@
...
 
-import net.minecraft.world.Container;
-import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.RecipeInput;
 import net.neoforged.neoforge.items.IItemHandlerModifiable;
 
-public class RecipeWrapper implements Container {
+public class RecipeWrapper implements RecipeInput {
     protected final IItemHandlerModifiable inv;
```
Comment (Technici4n) [inline src/main/java/net/neoforged/neoforge/items/wrapper/RecipeWrapper.java]: This could be a plain `IItemHandler`, right?

Comment (TelepathicGrunt) [reply src/main/java/net/neoforged/neoforge/items/wrapper/RecipeWrapper.java]: Good call. Yeah that is valid to do.

**Review by Technici4n — APPROVED** [review]
